samples/matter/bridge/src/bridged_device_types/onoff_light.cpp

```diff
@@ -0,0 +1,111 @@
+	app::Clusters::OnOff::Commands::Toggle::Id,
+	kInvalidCommandId,
+};
+
+DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(bridgedLightClusters)
+DECLARE_DYNAMIC_CLUSTER(Clusters::OnOff::Id, onOffAttrs, onOffIncomingCommands, nullptr),
+	DECLARE_DYNAMIC_CLUSTER(Clusters::Descriptor::Id, descriptorAttrs, nullptr, nullptr),
+	DECLARE_DYNAMIC_CLUSTER(Clusters::BridgedDeviceBasicInformation::Id, bridgedDeviceBasicAttrs, nullptr,
+				nullptr) DECLARE_DYNAMIC_CLUSTER_LIST_END;
+
+DECLARE_DYNAMIC_ENDPOINT(bridgedLightEndpoint, bridgedLightClusters);
+
+const EmberAfDeviceType gBridgedOnOffDeviceTypes[] = {
+	{ static_cast<chip::DeviceTypeId>(BridgedDevice::DeviceType::OnOffLight),
+	  BridgedDevice::kDefaultDynamicEndpointVersion },
+	{ static_cast<chip::DeviceTypeId>(BridgedDevice::DeviceType::BridgedNode),
+	  BridgedDevice::kDefaultDynamicEndpointVersion }
+};
+
+const uint8_t gLightDataVersionSize = ArraySize(bridgedLightClusters);
+
+OnOffLightDevice::OnOffLightDevice(const char *nodeLabel) : BridgedDevice(nodeLabel)
+{
+	mDataVersionSize = gLightDataVersionSize;
+	mEp = &bridgedLightEndpoint;
+	mDeviceTypeList = gBridgedOnOffDeviceTypes;
+	mDeviceTypeListSize = sizeof(gBridgedOnOffDeviceTypes);
+	mDataVersion =
+		reinterpret_cast<DataVersion *>(chip::Platform::MemoryAlloc(sizeof(DataVersion) * mDataVersionSize));
+}
+
+CHIP_ERROR OnOffLightDevice::HandleRead(ClusterId clusterId, AttributeId attributeId,
+					uint8_t *buffer, uint16_t maxReadLength)
+{
+	switch (clusterId) {
+	case Clusters::BridgedDeviceBasicInformation::Id:
+		return HandleReadBridgedDeviceBasicInformation(attributeId, buffer,
+									      maxReadLength);
+		break;
+	case Clusters::Descriptor::Id:
+		return HandleReadDescriptor(attributeId, buffer, maxReadLength);
+		break;
+	case Clusters::OnOff::Id:
+		return HandleReadOnOff(attributeId, buffer, maxReadLength);
+		break;
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+
+CHIP_ERROR OnOffLightDevice::HandleReadOnOff(AttributeId attributeId, uint8_t *buffer,
+					     uint16_t maxReadLength)
+{
+	switch (attributeId) {
+	case Clusters::OnOff::Attributes::OnOff::Id: {
+		bool onOff = GetOnOff();
+		return HandleReadAttribute(&onOff, sizeof(onOff), buffer, maxReadLength);
+	}
+	case Clusters::OnOff::Attributes::ClusterRevision::Id: {
+		uint16_t clusterRevision = GetOnOffClusterRevision();
+		return HandleReadAttribute(&clusterRevision, sizeof(clusterRevision), buffer, maxReadLength);
+	}
+	case Clusters::OnOff::Attributes::FeatureMap::Id: {
+		uint32_t featureMap = GetOnOffFeatureMap();
+		return HandleReadAttribute(&featureMap, sizeof(featureMap), buffer, maxReadLength);
+	}
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+
+CHIP_ERROR OnOffLightDevice::HandleWrite(ClusterId clusterId, AttributeId attributeId,
+					 uint8_t *buffer)
+{
+	if (clusterId != Clusters::OnOff::Id) {
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+
+	switch (attributeId) {
+	case Clusters::OnOff::Attributes::OnOff::Id: {
+		Set(*buffer);
```

Comment:
Missing `nullptr` check for `buffer`

Reply:
It's not, the check is done in `BridgedManager::HandleWrite` method, before invoking this one.

Reply:
Yes, but both `HandleWrite` and `GetBridgedDevice` are part of the public interface, so theoretically it's not guaranteed ConcreteDevice::HandleWrite will be always called in a way it is now :)

Reply:
I'm not quite convinced. To break thing up you would need to call `ConcreteDevice::HandleWrite` somewhere else, but how would you get a ConcreteDevice instance if it's a part of private array of the `BridgedManager`. The members of a list are get only using `GetBridgedDevice`, which handles argument checks correctly.